### ISSUE TITLE
Include current word-warp state in TextEditorOptions

### ISSUE THREAD

**Issue by OlegKarasik** [microsoft/vscode#70485]
Current the editor's `word-warp` settings are persisted in memory. Can this setting be included in `TextEditorOptions` and when changed trigger `onDidChangeTextEditorOptions` event?

The motivation for this is to have an ability to react on the `word-warp` setting changes in scenarios when you are mirroring active editor. 

My particular use case is in [html-escape](https://github.com/cfjedimaster/htmlescape-vscode) extension we create an editor to display escaped text from the active editor. The problem is when you change `word-warp` in active editor the what you see in our editor no longer corresponds to what you see in active editor (the user had to change `word-warp` in our editor too).

**Comment by alexandrudima** [comment]
@OlegKarasik Have you tried `window.onDidChangeTextEditorVisibleRanges` and `TextEditor.visibleRanges`. From what you are describing, they seem to do what you want.

**Comment by alexandrudima** [comment]
I am not understanding how this information would really help since it might at most cover the perfect case where the content is equal and the two editors are sized perfectly 50%-50%. It is always possible to resize split editors, for example the initial editor could have a width of 70% while the other one would have a width of 30%. So enabling word wrapping in both would not guarantee any kind of alignment.

This is especially true as the second editor is doing a kind of transformation for your extension. Even when opening the exact same file twice with word wrap or with folding regions, it is not possible to keep things perfectly in sync:

e.g. the same file as the same scroll top:
#### without word wrap lines could possibly be aligned
![image](https://user-images.githubusercontent.com/5047891/67872074-b4650780-fb31-11e9-8906-d2e35be33492.png)

#### with word wrap it is not possible to perfectly align things
![image](https://user-images.githubusercontent.com/5047891/67872119-c8a90480-fb31-11e9-9dbd-503a3fd6c21d.png)

---

It is also possible to fold certain regions in the source file... So I think misalignments are nothing you can avoid. 

In other words, I don't think this is a good enough reason for us to add API. I hope you don't mind, but we need to keep our API surface lean and slim in order to be able to maintain it...

**Comment by OlegKarasik** [comment]
Hi @alexandrudima,

I understand your point and I agree with it (I had an experience maintaining a framework, so I understand that sometimes you should say no to your consumers).

Still, I think I wasn't able to explain the core idea, so give me a last try to do this.

My point wasn't about trying to make mirror editor to exactly mirror content of source editor (as you've mentioned this isn't possible because of two reasons: they could have different widths and mirror editor mutates the content - in my case does HTML escapement). The core idea was in maintaining user experience.

Imagine a case: user works in source editor with word wrap turned on. The purpose of turning word wrap is to avoid horizontal scrolling. So when user creates a mirroring editor the basic expectation is that mirroring editor will have the word wrap turned on too (because if user don't want to have horizontal scrolling in source editor why would user want to have horizontal scrolling in mirroring editor?). The same applies to a situation when word wrap is dynamically changed is source editor.
